### PR TITLE
Added remark under a select of a profile to the HTML XCCDF report

### PR DESCRIPTION
**Pull request by Mab879**
This PR adds remark under a select of a profile as discussed in #1743.

Willing to fix any issues you all might find.

Big thanks to @cipherboy for pointing me to this issue.

### CODE REVIEW

**Comment by jan-cerny** [comment]
Thank you. It works great.

I have two suggestions:
1) It would be nice to have the same feature also in HTML guide. Sometimes the people review the HTML guides before applying the profile, so they would be interested in the remarks. It would also support https://github.com/ComplianceAsCode/content/issues/6889
2) I think it can be useful to enable it also for tailoring files - ie. for profiles that are children of `<xccdf:Tailoring>`. However it won't be as easy because you will have to pass the tailoring element as another parameter to the template.

File: xsl/xccdf-report-impl.xsl [hunk 1]
```diff
@@ -852,6 +852,11 @@ Authors:
                         </xsl:apply-templates>
                 </div></td></tr>
             </xsl:if>
+            <xsl:if test="$profile/cdf:select[@idref=$item/@id]/cdf:remark/text()">
```
Comment (cipherboy) [inline xsl/xccdf-report-impl.xsl:855]: Just an FYI, I was reading the standard tonight:

> The `<xccdf:select>` element MAY have one or more `<xccdf:remark>`
> elements (strings) containing explanatory material or other prose. Each
> instance of `<xccdf:remark>` MAY have an `@xml:lang` attribute (see
> Section 6.2.10).

IIUC, this doesn't handle having multiple remarks under the selected item. You'll probably need to iterate over each remark and add it to a separate section.

**Review by cipherboy — CHANGES_REQUESTED** [review]
Sadly, according to the XCCDF spec, we'll need to support multiple `<xccdf:remark />` elements and IIUC this doesn't yet. :/

File: xsl/xccdf-report-impl.xsl (hunk 1 above)
Comment (Mab879) [reply xsl/xccdf-report-impl.xsl]: This should fixed in my latest commit.

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Pull request merged**